app/models/comment.php

```diff
@@ -0,0 +1,55 @@
+<?php
+class Comment extends AppModel
+{
+    public $validation = array(
+        'body'          => array(
+            'length'    => array('validate_between', 1, 200),
+        ),
+    );
+
+    public function __construct(array $data = array())
+    {
+        parent::__construct($data);
+
+        if(!isset($this->user_id)) return;
+
+        $user = User::get($this->user_id);
+        $this->username = $user->username;
+    }
+
+    public static function countAll($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id={$thread_id}");
+    }
+
+    public static function getAll($thread_id, $offset, $limit)
+    {
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM comment WHERE thread_id={$thread_id} ORDER BY created ASC LIMIT {$offset}, {$limit}");
+
+        $comments = array();
+        foreach ($rows as $row) {
+            $comments[] = new Comment($row);
```

Comment:
You may change this to:
```$comments[] = new self($row);```, since you're already in the Comment model po.
Please apply to all po. :bow: